Handle invalid directory for incoming file transfers

infirit opened the pull request:
Fixes #998

cschramm commented:
Shouldn't we better ask the user for confirmation before resetting the configuration instead of telling him that we just did that? :smile:

zeehio commented:
At least for the case reported on #998 I would not expect any confirmation (no other software asks for confirmation in that case).

For other cases I don't know.

infirit commented:
I don't mind asking the user but the transfer will fail if they don't act. So instead I now have,

* A function construct the share path which should be valid, return tuple with path and if stored path was valid
* Use the new function instead of retrieving the path from gsettings
* Show a notification informing the user the stored path is invalid and what we use instead

Updated the commit for comment but is has not been tested yet (which i'll do later).

cschramm commented:
@zeehio: Right, doesn't make much sense for #998.

@infirit: :+1: I like the solution.

infirit commented:
@zeehio yeah, the problem I have to deal with is that all users up till now (which includes you) will have the full path stored in out settings. So I took a bit of a (too) drastic measure.

Your comment made me think. I can probably compare the stored path with our default and reset if they are the same. I am very tempted to add this :thinking:

infirit commented:
Added the reset when stored and default path are identical. Will merge in a couple of days unless someone has objections.

zeehio commented:
It is admirable that you are putting so much effort to address this properly. Thank you 🙂 and sorry for wasting your time 😅.

If on new installations everything works fine, then in the long term the problem will be solved for everyone and if I were you I would not put much more work into it, but of course do as you please :-)

In the GUI blueman-services, where I am able to choose the download path, would you think it is worth to give the option "Use the system default" vs "Specify a custom directory"? New users "Use the system default" while users coming from older versions would have a "custom directory" set, unless your path comparison trick has reset the setting.

Ideally, if the custom directory does not exist, the user would be warned and ideally he/she would be given the option to restore to the system default (informing what the default is) in the same dialog. In my case I did not even know what blueman-services was when I saw the message and I had to stop and think how to open blueman-services. If I had been given the option "restore to system default" I would have clicked it almost without thinking. Or at least "Click here to open blueman-services" would have felt more natural than "find blueman-services and open it".

If I think about the user who does not know how to open blueman-services, having a notification on each login about some invalid path is less annoying than an error message, because the notification disappears without having to click it. However, having a dialog only once that allows me to solve the issue ("click here to use the default setting") would be much better than the notification on every login.

In any case, thanks again for your work. This PR solves the issue and blueman-services were actually easy to find when I got tired of closing the error message (in my case I made the effort of looking for blueman-services after three days clicking once every day). 😅

cschramm commented on blueman/plugins/applet/TransferService.py:
```diff
@@ -171,6 +161,32 @@ def on_unload(self):
+        path = None
+        error = False
+
+        if config_path == '':
```
Is this true for the initial configuration or will querying `shared-path` give us `None` instead? :thinking:

infirit replied:
Yes, the default in the schema is an [empty string](https://github.com/blueman-project/blueman/blob/master/data/org.blueman.gschema.xml#L159).

infirit commented:
@zeehio no need to apologise and you are not. I added an action (think button) to the notification that will allow the user to reset. The notification should stay up for 30 seconds and then go away.

@cschramm also added a commit to fix the open action_cb on completed file transfers.
